src/components/CreateAccount/CreateAccount.tsx

```diff
@@ -1,8 +1,297 @@
+  padding: 16px;
+  border-radius: 12px;
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+`;
+
+const InputContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+
+  label {
+    font-size: 12px;
+    font-weight: 500;
+  }
+
+  input {
+    padding: 8px 12px;
+    border: 1px solid #e5e5e5;
+    border-radius: 10px;
+    font-size: 14px;
+    margin-top: 4px;
+    min-height: 50px;
+    cursor: text;
+
+    &:focus {
+      outline: none;
+      border: 1px solid #e5e5e5;
+    }
+  }
+
+  .subText {
+    font-size: 0.75rem;
+    padding: 8px 0;
+
+    .error {
+      color: hsla(8, 100%, 33%, 1);
+    }
+
+    .success {
+      color: hsla(155, 66%, 32%, 1);
+    }
+  }
+`;
 
 function CreateAccount() {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    formState: { errors, touchedFields },
+    clearErrors,
+  } = useForm();
+  const formValues = watch();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+
+  const createAccount = async (data: { email: string; username: string; }) => {
+    const success_url = searchParams.get('success_url');
+    const failure_url = searchParams.get('failure_url');
+    const public_key =  searchParams.get('public_key');
+    const contract_id = searchParams.get('contract_id');
+    const methodNames = searchParams.get('methodNames');
+    try {
+      const fullAccountId = `${data.username}.${network.fastAuth.accountIdSuffix}`;
+      const {
+        publicKey: publicKeyFak, email, privateKey, accountId
+      } = await handleCreateAccount({
+        accountId:   fullAccountId,
+        email:       data.email,
+        isRecovery:  false,
+        success_url,
+        failure_url,
+        public_key,
+        contract_id,
+        methodNames,
+      });
+      const newSearchParams = new URLSearchParams({
+        accountId,
+        publicKeyFak,
+        email,
+        isRecovery: 'false',
+        ...(success_url ? { success_url } : {}),
+        ...(failure_url ? { failure_url } : {}),
+        ...(public_key ? { public_key_lak: public_key } : {}),
+        ...(contract_id ? { contract_id } : {}),
+        ...(methodNames ? { methodNames } : {})
+      });
+      const hashParams = new URLSearchParams({ privateKey });
+      navigate(`/verify-email?${newSearchParams.toString()}#${hashParams.toString()}`);
+    } catch (error: any) {
+      openToast({
+        type:  'ERROR',
+        title: error.message,
+      });
+    }
+  };
+
+  useEffect(() => {
+    const checkPassKey = async (): Promise<void> => {
+      const isPasskeyReady = await isPassKeyAvailable();
+      if (!isPasskeyReady) {
+        openToast({
+          title: '',
+          type:  'INFO',
+          description:
+            'Passkey support is required for account creation. Try using an updated version of Chrome or Safari to create an account.',
+          duration: 5000,
+        });
+      }
+    };
+    checkPassKey();
```

Comment:
I believe this is no longer there in the latest version of fast auth - we are just creating accounts with LAKs in this scenario so we would redirect to `successUrl` with added LAK in this case

Reply:
I don't understand this one. Do I just remove this code?

Reply:
We will address this as a separate issue, created #35